### PR TITLE
DIT-879 show assignee or assigned queue

### PR DESCRIPTION
**Pull request by retroredge**

### CODE REVIEW

File: app/uk/gov/hmrc/tariffclassificationfrontend/models/Queues.scala [hunk 1]
```diff
@@ -22,4 +22,12 @@ object Queues {
   val cap = Queue("3", "cap", "CAP")
   val cars = Queue("4", "cars", "Cars")
   val elm = Queue("5", "elm", "ELM")
+
+  val queueNames: Map[String, String] = Map(
```
Comment (gokyo) [inline app/uk/gov/hmrc/tariffclassificationfrontend/models/Queues.scala]: `object Queues` has redundant definitions

Comment (SteveSugden) [reply app/uk/gov/hmrc/tariffclassificationfrontend/models/Queues.scala]: @pllaurado - we will be able to use this map of queueId to name to display our new event

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html [hunk 2]
```diff
@@ -144,18 +145,36 @@
...
-                "This case is currently unassigned"
+                <span id="activity-events-assigned-queue-label">
+                    @messages("case.activity.assigned-queue")
+                </span>
+                <span id="activity-events-assigned-queue">
+                @{
+                    if(c.queueId.isDefined) {
+                        Queues.queueNames(c.queueId.get)
+                    } else {
+                        "Gateway"
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html]: Just checking that it's ok to hard-code the Gateway name here.

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html [hunk 3]
```diff
@@ -144,18 +145,36 @@
...
+        <div class="column-full">
+            @if(c.assignee.isDefined) {
+                <span id="activity-events-assignee-label">
+                    @messages("case.activity.assigned-to")
+                </span>
+                <span id="activity-events-assignee">
+                @{
+                    if(c.assignee.map(_.id).contains(request.operator.id)) {
+                        "you"
+                    } else if(c.assignee.flatMap(_.name).isDefined) {
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html]: Would assignee.scala.html help here?

File: app/uk/gov/hmrc/tariffclassificationfrontend/models/Queues.scala (hunk 1 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/models/Queues.scala]: Any reason why this is hard coded? It'd be quite easy to generate

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/models/Queues.scala]: Also this is the kind of thing the Queue Service is for

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html (hunk 3 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html]: Yes I think the inside of the ` @if(c.assignee.isDefined){...}` block could be replaced with `@assignee(c.assignee.get)`

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html [hunk 4]
```diff
@@ -144,18 +145,36 @@
...
+                </span>
             } else {
-                "This case is currently unassigned"
+                <span id="activity-events-assigned-queue-label">
+                    @messages("case.activity.assigned-queue")
+                </span>
+                <span id="activity-events-assigned-queue">
+                @{
+                    if(c.queueId.isDefined) {
+                        Queues.queueNames(c.queueId.get)
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html]: I'd rather this came via the queue service somehow if possible

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html (hunk 2 above)
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html]: You could use `Queues.gateway.name`

File: test/unit/uk/gov/hmrc/tariffclassificationfrontend/views/partials/ActivityDetailsViewSpec.scala [hunk 5]
```diff
@@ -322,22 +324,40 @@ class ActivityDetailsViewSpec extends ViewSpec {
...
+        withAssignee(None)
+      )
+
+      // When
+      val doc = view(activity_details(c, Paged.empty[Event], ActivityForm.form))
+
+      // Then
+      doc should containElementWithID("activity-events-assigned-queue")
+      doc.getElementById("activity-events-assigned-queue").text() shouldBe "Gateway"
+      doc.getElementById("activity-events-assigned-queue-label").text() shouldBe "Currently in:"
```
Comment (ed0906) [inline test/unit/uk/gov/hmrc/tariffclassificationfrontend/views/partials/ActivityDetailsViewSpec.scala]: I'm not sure we need the `activity-events-assigned-queue-label` assertion as theres nothing dynamic in there

File: app/uk/gov/hmrc/tariffclassificationfrontend/models/Queues.scala (hunk 1 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/models/Queues.scala]: Service now provides the queue name

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html (hunk 3 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html]: Seems to work as expected

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html (hunk 4 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/activity_details.scala.html]: Done

File: app/uk/gov/hmrc/tariffclassificationfrontend/service/QueuesService.scala [hunk 6]
```diff
@@ -25,6 +25,10 @@ class QueuesService {
 
   def getAll: Seq[Queue] = Seq(gateway, act, cap, cars, elm)
 
+  def queueNameForId(id: String): String = {
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/service/QueuesService.scala]: You could have just got the full Queue and used the existing methods

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/service/QueuesService.scala]: Then do `queue.name` in the view

Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/service/QueuesService.scala]: OK. The `find` will return an `Option[Queue]` and the view can deal with that

**Pull request merged**